diktat-rules/src/test/kotlin/org/cqfn/diktat/ruleset/chapter3/CollapseIfStatementsRuleWarnTest.kt

```diff
@@ -0,0 +1,281 @@
+            |                     }
+            |                 }
+            |             }    
+            |         }
+            |     }    
+            |}
+            """.trimMargin(),
+            LintError(3, 10, ruleId, "${COLLAPSE_IF_STATEMENTS.warnText()} avoid using redundant nested if-statements", true),
+            LintError(4, 14, ruleId, "${COLLAPSE_IF_STATEMENTS.warnText()} avoid using redundant nested if-statements", true),
+            LintError(5, 18, ruleId, "${COLLAPSE_IF_STATEMENTS.warnText()} avoid using redundant nested if-statements", true),
+            LintError(6, 22, ruleId, "${COLLAPSE_IF_STATEMENTS.warnText()} avoid using redundant nested if-statements", true),
+            LintError(7, 26, ruleId, "${COLLAPSE_IF_STATEMENTS.warnText()} avoid using redundant nested if-statements", true)
+        )
+    }
+
+    @Test
+    @Tag(WarningNames.COLLAPSE_IF_STATEMENTS)
+    fun `else if statement`() {
+        lintMethod(
+            """
+            |fun foo() {
+            |    fun1()
+            |    if (cond1) {
+            |        fun2()
+            |    } else if (cond2) {
+            |        if (true) {
+            |            fun3()
+            |        }
+            |    } else {
+            |        fun4()
+            |    }
+            |}
+            """.trimMargin(),
+            LintError(6, 9, ruleId, "${COLLAPSE_IF_STATEMENTS.warnText()} avoid using redundant nested if-statements", true),
+        )
+    }
+
+    @Test
+    @Tag(WarningNames.COLLAPSE_IF_STATEMENTS)
+    fun `else if statement 2`() {
+        lintMethod(
+            """
+            |fun foo() {
+            |    fun1()
+            |    if (cond1) {
+            |        fun2()
+            |    } else if (cond2) {
+            |        if (true) {
+            |           if (true) {
+            |               fun3()
+            |           } 
+            |        }
+            |    } else {
+            |        fun4()
+            |    }
+            |}
+            """.trimMargin(),
+            LintError(6, 9, ruleId, "${COLLAPSE_IF_STATEMENTS.warnText()} avoid using redundant nested if-statements", true),
+            LintError(7, 12, ruleId, "${COLLAPSE_IF_STATEMENTS.warnText()} avoid using redundant nested if-statements", true),
+        )
+    }
+
+    @Test
+    @Tag(WarningNames.COLLAPSE_IF_STATEMENTS)
+    fun `compound condition`() {
+        lintMethod(
+            """
+            |fun foo () {
+            |     if (cond1) {
+            |         if (cond2 || cond3) {
+            |             firstAction()
+            |             secondAction()
+            |         }
+            |         if (cond4) {
+            |             secondAction()
+            |         }
+            |     }
+            |} 
+            """.trimMargin(),
+            LintError(3, 10, ruleId, "${COLLAPSE_IF_STATEMENTS.warnText()} avoid using redundant nested if-statements", true)
```

Comment:
I don't think this is a correct example: we should either emit warnings for both nested ifs or for none of them, because rewriting this would mean that `cond4` should be merged with `cond1` too

Reply:
Very good remark, thank you, seems for now it's fixed